### PR TITLE
Remove 6.0.* version constants

### PR DESCRIPTION
**Pull request by nknize**
This PR removes LegacyESVersion.V_6_0_* constants including all pre-release
versions and bug fixes in preparation for 2.0.0 and Lucene 9 upgrade.

### CODE REVIEW

**Comment by nknize** [comment]
start gradle check

**Comment by nknize** [comment]
Previous check failure giving an `AccessDenied` on the log. I wonder if this was a glitch in the jenkins matrix.

**Comment by nknize** [comment]
[AWS Status](https://status.aws.amazon.com/) is currently affecting PR builds. Validation awaits resolution...

**Comment by nknize** [comment]
start gradle check

**Comment by nknize** [comment]
start gradle check

**Comment by nknize** [comment]
start gradle precommit

**Comment by dblock** [comment]
I think we prematurely disabled pre-commit before merging #1664.

**Review by adnapibar — COMMENTED** [review]
Thanks Nick! LGTM - just a few comments.

File: server/src/test/java/org/opensearch/action/admin/cluster/health/ClusterHealthResponsesTests.java [hunk 1]
```diff
@@ -185,7 +184,7 @@ public void testVersionCompatibleSerialization() throws IOException {
         );
 
         BytesStreamOutput out_lt_1_0 = new BytesStreamOutput();
-        Version old_version = VersionUtils.randomVersionBetween(random(), LegacyESVersion.V_6_0_0, LegacyESVersion.V_6_8_0);
+        Version old_version = VersionUtils.randomIndexCompatibleVersion(random());
```
[the remark this answers is not in the document]
Comment (nknize) [reply server/src/test/java/org/opensearch/action/admin/cluster/health/ClusterHealthResponsesTests.java:187]: :+1:  Nice catch! Fixed.

File: server/src/test/java/org/opensearch/cluster/coordination/JoinTaskExecutorTests.java [hunk 2]
```diff
@@ -119,21 +118,8 @@ public void testPreventJoinClusterWithUnsupportedNodeVersions() {
...
-            expectThrows(IllegalStateException.class, () -> {
-                if (randomBoolean()) {
-                    JoinTaskExecutor.ensureNodesCompatibility(tooHigh, nodes);
-                } else {
-                    JoinTaskExecutor.ensureNodesCompatibility(tooHigh, minNodeVersion, maxNodeVersion);
-                }
-            });
-        }
-
-        if (minNodeVersion.onOrAfter(LegacyESVersion.V_7_0_0)) {
```
[the remark this answers is not in the document]
Comment (nknize) [reply server/src/test/java/org/opensearch/cluster/coordination/JoinTaskExecutorTests.java:133]: Yes! Or probably when we remove the V_6_4_0 version is fine.

[the remark this answers is not in the document]
Comment (nknize) [reply server/src/test/java/org/opensearch/cluster/coordination/JoinTaskExecutorTests.java:133]: reverted this change

File: server/src/main/java/org/opensearch/plugins/PluginInfo.java [hunk 3]
```diff
@@ -174,13 +174,6 @@ public PluginInfo(final StreamInput in) throws IOException {
             extendedPlugins = Collections.emptyList();
         }
         hasNativeController = in.readBoolean();
-        if (in.getVersion().onOrAfter(LegacyESVersion.V_6_0_0_beta2) && in.getVersion().before(LegacyESVersion.V_6_3_0)) {
```
[the remark this answers is not in the document]
Comment (nknize) [reply server/src/main/java/org/opensearch/plugins/PluginInfo.java:177]: yes! :+1:

**Comment by nknize** [comment]
Le Sigh... Azure timeouts!!

**Comment by nknize** [comment]
start gradle check

**Comment by adnapibar** [comment]
Looks like the transient timeouts - retrying check

```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':qa:smoke-test-multinode:integTest'.
> `cluster{:qa:smoke-test-multinode:integTest}` failed to wait for cluster health yellow after 40 SECONDS
    IO error while waiting cluster
    503 Service Unavailable
```

**Comment by adnapibar** [comment]
start gradle check

**Review by adnapibar — APPROVED** [review]
👍

**Pull request merged**